docs/development/extensions-contrib/iceberg.md

```diff
@@ -0,0 +1,121 @@
+
+### Load the Iceberg Ingest extension
+
+To use the iceberg extension, add the `druid-iceberg-extensions` to the list of loaded extensions. See [Loading extensions](../../configuration/extensions.md#loading-extensions) for more information.
+
+
+### Hive Metastore catalog
+
+For Druid to seamlessly talk to the Hive Metastore, ensure that the Hive specific configuration files such as `hive-site.xml` and `core-site.xml` are available in the Druid classpath for peon processes.  
+Hive specific properties can also be specified under the `catalogProperties` object in the ingestion spec. 
+
+Hive metastore catalogs can be associated with different types of warehouses, but this extension presently only supports HDFS and S3 warehouse directories.
+
+#### Reading from HDFS warehouse 
+
+Ensure that the extension `druid-hdfs-storage` is loaded. The data file paths are extracted from the Hive metastore catalog and [HDFS input source](../../ingestion/input-sources.md#hdfs-input-source) is used to ingest these files.
+The `warehouseSource` type in the ingestion spec should be `hdfs`.
+
+If the Hive metastore supports Kerberos authentication, the following properties will be required in the `catalogProperties`:
+
+```json
+"catalogProperties": {
+  "principal": "krb_principal",
+  "keytab": "/path/to/keytab"
+}
+```
+Only Kerberos based authentication is supported as of now.
+
+#### Reading from S3 warehouse
+
+Ensure that the extension `druid-s3-extensions` is loaded. The data file paths are extracted from the Hive metastore catalog and `S3InputSource` is used to ingest these files.
+The `warehouseSource` type in the ingestion spec should be `s3`. If the S3 endpoint for the warehouse is different from the endpoint configured as the deep storage, the following properties are required in the `warehouseSource` section to define the S3 endpoint settings:
+
+```json
+"warehouseSource": {
+  "type": "s3",
+  "endpointConfig": {
+    "url": "S3_ENDPOINT_URL",
+    "signingRegion": "us-east-1"
+  },
+  "clientConfig": {
+    "protocol": "http",
+    "disableChunkedEncoding": true,
+    "enablePathStyleAccess": true,
+    "forceGlobalBucketAccessEnabled": false
+  },
+  "properties": {
+    "accessKeyId": {
+      "type": "default",
+      "password": "<ACCESS_KEY_ID"
+    },
+    "secretAccessKey": {
+      "type": "default",
+      "password": "<SECRET_ACCESS_KEY>"
+    }
+  }
+}
+```
+
+This extension uses the [Hadoop AWS module](https://hadoop.apache.org/docs/stable/hadoop-aws/tools/hadoop-aws/) to connect to S3 and retrieve the metadata and data file paths.
+The following properties will be required in the `catalogProperties`:
+
+```json
+"catalogProperties": {
+  "fs.s3a.access.key" : "S3_ACCESS_KEY",
+  "fs.s3a.secret.key" : "S3_SECRET_KEY",
+  "fs.s3a.endpoint" : "S3_API_ENDPOINT"
+}
+```
+Since the Hadoop AWS connector uses the `s3a` filesystem based client, the warehouse path should be specified with the `s3a://` protocol instead of `s3://`.
```

Comment:
```suggestion
Since the Hadoop AWS connector uses the `s3a` filesystem client, specify the warehouse path with the `s3a://` protocol instead of `s3://`.
```